aw_transform/heartbeats.py

```diff
@@ -25,14 +25,21 @@ def heartbeat_reduce(events: List[Event], pulsetime: float) -> List[Event]:
 
 def heartbeat_merge(last_event: Event, heartbeat: Event, pulsetime: float) -> Optional[Event]:
     """
-    Merges two events if they have identical labels and are
-    separated by a time smaller than :code:`pulsetime` seconds.
+    Merges two events if they have identical data and are either:
+     - Separated by a time smaller than :code:`pulsetime` seconds.
+     - Sharing the same starting timestamp.
     """
     if last_event.data == heartbeat.data:
+        # Seconds between end of last_event and start of heartbeat
         gap = heartbeat.timestamp - (last_event.timestamp + last_event.duration)
+        within_pulsetime_window = gap <= timedelta(seconds=pulsetime)
 
-        if gap <= timedelta(seconds=pulsetime):
-            # Heartbeat was within pulsetime window, set duration of last event appropriately
+        same_start = heartbeat.timestamp == last_event.timestamp
+
+        # First condition checks if within pulsetime window
+        # Second condition checks if the new event is just an updated version of the last one
+        if within_pulsetime_window or same_start:
```

Comment:
Why is same_start needed? Shouldn't gap be enough?
Seems like same_start should only be an issue if the watcher is badly written and doesn't report within it's on pulsetime window.

Reply:
> Should we still keep this change? It does no real harm.

True, but IMO it's better to just keep it as is to make the heartbeat implementation simpler. Feels a bit inconsistent that there are two types of heartbeats techniques in one implementation.

Reply:
Another way to do it, that's arguably cleaner, would be to check if the heartbeat has its timestamp within the timeperiod `(last_event.timestamp) -> (last_event.timestamp + last_event.duration + pulsetime)`. Would make a lot of sense to me tbh.

Reply:
Hm, that sounds pretty decent actually.
I'd prefer to think about it a bit more though, not entirely sure.

Reply:
Implemented it in https://github.com/ActivityWatch/aw-core/pull/59/commits/5f640520bd0747570d4c63298284606d2ac159b4